### ISSUE TITLE
allow setting headers on probe request

### ISSUE THREAD

**Issue by MarcoDuiker** [geopython/GeoHealthCheck#186]
In my experience it is often needed to set specific headers on a request (eg. WFS GET or POST). I ran into such an issue right now.

It would be nice (necessary?) if the option to add headers to a request would be added to the relevant probes.

**Comment by justb4** [comment]
Good point. I need this as well. Some Probes e.g. WFS and HTTP have a`REQUEST_HEADERS` param, but only for `content-type`. Need to think how to add in a generic fashion. You have suggestions?

**Comment by MarcoDuiker** [comment]
Just (no pun intended) add the possibility to add an arbitrary number of arbitrary key-value pairs to send as headers. That should suffice.

**Comment by justb4** [comment]
@MarcoDuiker was your use case related to Authentication? Then you may try latest `master` version or via https://demo.geohealthcheck.org, via #9 and #259 per-Resource authentication via `Authorization` HTTP header has been implemented.

But we keep this issue as there may be other use cases off course.

**Comment by MarcoDuiker** [comment]
Nope, not related to authentication. Indeed another use case.

**Comment by tobias-brunner** [comment]
This would indeed be a welcomed feature. We need custom headers like "api-key", as some serviced are authenticated in this fashion.

So a list with key-value pairs as @MarcoDuiker suggested would be the definitive solution.
